### PR TITLE
fix(api): reject requests with invalid parameters

### LINKED ISSUE #210: Reject unexpected parameters on most endpoints

### PR DESCRIPTION
**Pull request by seanmonstar**
This keeps the leniency for the POST /v1/token endpoint, but
cactches errors for misuses of other endpoints.

Closes #210 

@washort sanity checking, this doesn't break you, does it? It's only the token endpoint that receives extra parameters?

### CODE REVIEW

File: test/api.js [hunk 1]
```diff
@@ -157,7 +157,7 @@ describe('/v1', function() {
     describe('GET', function() {
       it('redirects with all query params', function(done) {
         Server.api
-        .get('/authorization?client_id=123&state=321&scope=1&action=signup')
+        .get('/authorization?client_id=123&state=321&scope=1&action=signup&a=b')
```
Comment (rfk) [inline test/api.js]: Maybe worth adding  a comment about what the `a=b` is testing for, and/or a comment as to why we allow `stripUnknown` in the /token case.

**Comment by rfk** [comment]
r+ with a single nit

**Comment by ckarlof** [comment]
So the outcome here is that `GET /authorization` still forwards unanticipated params, but others block?

**Comment by seanmonstar** [comment]
This didn't actually affect the `GET` request, as there was no validation on that route. It's the `POST /token` request that needed that (and so uses `stripUnknown`). All other endpoints will error with a 400.

**Comment by ckarlof** [comment]
Gotcha.

**Pull request merged**